<?xml version="1.0"?>

<launch>


	<!-- consumer -->
	<arg name="SERVICE_NAME_CONSUMER" default="temperature_c"/>
	<arg name="UNIT_CONSUMER" default="Celsius"/>
	<arg name="TARGET_SYSTEM_NAME_CONSUMER" default="temperature_provider_demo"/>
	<arg name="TARGET_ADDRESS_CONSUMER" default="130.240.5.68"/>
	<arg name="TARGET_PORT_CONSUMER" default="8490"/>
	<!-- publisher -->
	<arg name="SERVICE_NAME_PUBLISHER" default="temperature_f"/>
	<arg name="UNIT_PUBLISHER" default="Fahrenheit"/>
	<arg name="TARGET_SYSTEM_NAME_PUBLISHER" default="temperature_subscriber_demo"/>
	<arg name="TARGET_ADDRESS_PUBLISHER" default="130.240.5.68"/>
	<arg name="TARGET_PORT_PUBLISHER" default="8494"/>
	<!-- booth -->	
	<arg name="ACCESS_URI" default="http://localhost:8440/orchestrator/orchestration"/>
	<arg name="ACCESS_URI_HTTPS" default="https://localhost:8441/orchestrator/orchestration"/>
	<arg name="THIS_ADDRESS" default="130.240.5.68"/>
	<arg name="THIS_ADDRESS6" default="[fe80::45a5:38c:f4d4:8ede]"/>
	<arg name="THIS_SYSTEM_NAME" default="temperature_consumer_demo"/>
	<arg name="INTERFACE" default="JSON"/>
	<arg name="SECURITY" default="Token"/>
	<arg name="THIS_PORT" default="8492"/>
	<arg name="SECURE_ARROWHEAD_INTERFACE" default="false"/>
	<arg name="SECURE_PROVIDER_INTERFACE" default="false"/>
	<arg name="OVERRIDE_STORE" default="true"/>
	<arg name="MATCHMAKING" default="true"/>
	<arg name="METADATA_SEARCH" default="false"/>
	<arg name="PING_PROVIDERS" default="false"/>
	<arg name="ONLY_PREFERRED" default="true"/>
	<arg name="EXTERNAL_SERVICE_REQUEST" default="false"/>



	<!-- service -->
	<node pkg="consumer_publisher" type="consumer_publisher" name="consumer_publisher">
		<param name="PRIVATE_KEY_PATH" value="Absolute_path_to_private_key"/>
		<param name="PUBLIC_KEY_PATH" value="Absolute_path_to_public_key"/>
		<!-- consumer -->
		<param name="SERVICE_NAME_CONSUMER" value="$(arg SERVICE_NAME_CONSUMER)"/>
		<param name="UNIT_CONSUMER" value="$(arg UNIT_CONSUMER)"/>
		<param name="TARGET_SYSTEM_NAME_CONSUMER" value="$(arg TARGET_SYSTEM_NAME_CONSUMER)"/>
		<param name="TARGET_ADDRESS_CONSUMER" value="$(arg TARGET_ADDRESS_CONSUMER)"/>
		<param name="TARGET_PORT_CONSUMER" value="$(arg TARGET_PORT_CONSUMER)"/>
		<!-- publisher -->
		<param name="SERVICE_NAME_PUBLISHER" value="$(arg SERVICE_NAME_PUBLISHER)"/>
		<param name="UNIT_PUBLISHER" value="$(arg UNIT_PUBLISHER)"/>
		<param name="TARGET_SYSTEM_NAME_PUBLISHER" value="$(arg TARGET_SYSTEM_NAME_PUBLISHER)"/>
		<param name="TARGET_ADDRESS_PUBLISHER" value="$(arg TARGET_ADDRESS_PUBLISHER)"/>
		<param name="TARGET_PORT_PUBLISHER" value="$(arg TARGET_PORT_PUBLISHER)"/>
		<!-- booth -->	
		<param name="ACCESS_URI" value="$(arg ACCESS_URI)"/>
		<param name="ACCESS_URI_HTTPS" value="$(arg ACCESS_URI_HTTPS)"/>
		<param name="THIS_ADDRESS" value="$(arg THIS_ADDRESS)"/>
		<param name="THIS_ADDRESS6" value="$(arg THIS_ADDRESS6)"/>
		<param name="THIS_SYSTEM_NAME" value="$(arg THIS_SYSTEM_NAME)"/>
		<param name="INTERFACE" value="$(arg INTERFACE)"/>
		<param name="SECURITY" value="$(arg SECURITY)"/>
		<param name="THIS_PORT" value="$(arg THIS_PORT)"/>
		<param name="SECURE_ARROWHEAD_INTERFACE" value="$(arg SECURE_ARROWHEAD_INTERFACE)"/>
		<param name="SECURE_PROVIDER_INTERFACE" value="$(arg SECURE_PROVIDER_INTERFACE)"/>
		<param name="OVERRIDE_STORE" value="$(arg OVERRIDE_STORE)"/>
		<param name="MATCHMAKING" value="$(arg MATCHMAKING)"/>
		<param name="METADATA_SEARCH" value="$(arg METADATA_SEARCH)"/>
		<param name="PING_PROVIDERS" value="$(arg PING_PROVIDERS)"/>
		<param name="ONLY_PREFERRED" value="$(arg ONLY_PREFERRED)"/>
		<param name="EXTERNAL_SERVICE_REQUEST" value="$(arg EXTERNAL_SERVICE_REQUEST)"/>
	</node>


</launch>

